digits/dataset/generic/views.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2016, NVIDIA CORPORATION.  All rights reserved.
+from __future__ import absolute_import
+
+import flask
+
+from .forms import GenericDatasetForm
+from .job import GenericDatasetJob
+
+from digits import dataset, extensions, model, utils
+from digits.utils.routing import request_wants_json, job_from_request
+from digits.webapp import app, scheduler
+
+blueprint = flask.Blueprint(__name__, __name__)
+
+@blueprint.route('/new/<extension_id>', methods=['GET'])
+@utils.auth.requires_login
+def new(extension_id):
+    """
+    Returns a form for a new GenericDatasetJob
+    """
+
+    form = GenericDatasetForm()
+
+    ## Is there a request to clone a job with ?clone=<job_id>
+    utils.forms.fill_form_if_cloned(form)
+
+    extension = extensions.data.get_extension(extension_id)
+    extension_form = extension.get_dataset_form()
+
+    ## Is there a request to clone a job with ?clone=<job_id>
+    utils.forms.fill_form_if_cloned(extension_form)
+
+    template, context = extension.get_dataset_template(extension_form)
+    rendered_extension = flask.render_template_string(template, **context)
+
+    return flask.render_template('datasets/generic/new.html',
+        extension_title = extension.get_title(),
+        extension_id = extension_id,
+        extension_html = rendered_extension,
+        form=form
```

Comment:
Spacing here doesn't match other arguments. According to `flake8`, there shouldn't be any spaces around the `=` in keyword arguments. We haven't done that in the rest of DIGITS, but we might as well start following the convention when it's convenient.